api/envoy/config/listener/v3/listener.proto

```diff
@@ -106,7 +106,8 @@ message Listener {
   // The address that the listener should listen on. In general, the address must be unique, though
   // that is governed by the bind rules of the OS. E.g., multiple listeners can listen on port 0 on
   // Linux as the actual port will be allocated by the OS.
-  core.v3.Address address = 2 [(validate.rules).message = {required: true}];
+  // Required unless *api_listener* or *listener_specifier* is populated.
+  core.v3.Address address = 2;
```

Comment:
Should this be promoted to `oneof` in the future?
If so, we should move the other fields next to this one and add the `oneof_promotion` tag next to them.

Also this should probably also be reflected in the code. Looking at the [ListenerImpl c'tor](https://github.com/envoyproxy/envoy/blob/0b804a8d78f6afc29f8da7bcba4473a7d85dd1eb/source/server/listener_impl.cc#L300), it seems that the code will reject this config.

Reply:
See https://github.com/envoyproxy/envoy/issues/15372 for the plan on eventually converting this to a `oneof`.  It will wind up being restructured to make that happen.

Envoy does not currently support non-socket listeners, so the address field is required for Envoy.  If the code rejects that config, then that sounds like the right behavior to me.